industry_benchmarks/utils/fix_networks.py

```diff
@@ -0,0 +1,605 @@
+from openfe import ProteinComponent
+from openfe.protocols.openmm_rfe.equil_rfe_methods import (
+    RelativeHybridTopologyProtocol
+)
+
+from openfe.setup import KartografAtomMapper, lomap_scorers
+
+
+def parse_alchemical_network(
+    input_alchem_network_json_path: str,
+) -> AlchemicalNetwork:
+    """
+    Read an AlchemicalNetwork
+    """
+    j_dict = json.load(
+        open(input_alchem_network_json_path, "r"), cls=JSON_HANDLER.decoder
+    )
+    alchem_network = AlchemicalNetwork.from_dict(j_dict)
+    return alchem_network
+
+
+def _get_check_results_json(filename: str):
+    """
+    Get a json file, open it, check it's
+    a results json, and return it.
+    """
+    if not os.path.exists(filename):
+        print(f"Error: {filename} does not exist.")
+        return None
+
+    ru = json.load(open(filename, 'rb'), cls=JSON_HANDLER.decoder)
+
+    if ru.get("__qualname__") == "AlchemicalNetwork":
+        print(f"{filename} is a network_setup.json, skipping")
+        return None
+    if ru.get("__qualname__") == "Transformation":
+        print(f"{filename} is an input json, skipping")
+        return None
+    if "unit_results" not in ru.keys():
+        print(f"{filename} has no unit results - likely a failure")
+        return None
+    if any('exception' in u for u in ru['unit_results'].values()):
+        print(f"{filename} is a failed simulation")
+        return None
+
+    return ru
+
+
+def get_transformation_alternate(
+    pur: gufe.ProtocolUnitResult, ligand_network: LigandNetwork
+):
+    """
+    Getting a transformation if things got deleted.
+    """
+    # The way this was being done in PR #144 doesn't work because
+    # it would lead to either a solvent or vacuum simulation, not a
+    # complex one!
+    errmsg = ("The transformation data is not available, "
+              "please contact the OpenFE team")
+    raise ValueError(errmsg)
+
+
+def get_transformation(pur: gufe.ProtocolUnitResult):
+    """
+    Get a transformation out of a PUR
+
+    Returns
+    -------
+    transform : gufe.Transformation
+      The transformation for this edge
+    phase : str
+      Either "complex" or "solvent" depending on the phase type.
+    """
+    # We a assume a single result
+    ru_keys = [k for k in pur["protocol_result"]["data"].keys()]
+    if len(ru_keys) > 1:
+        errmsg = "Too many keys in the Protocol Unit Result file"
+        raise ValueError(errmsg)
+
+    units = pur["protocol_result"]["data"][ru_keys[0]]
+
+    if len(units) > 1:
+        errmsg = "Too many units found"
+        raise ValueError(errmsg)
+
+    if "stateA" not in units[0]["inputs"]:
+        return None, None
+
+    stateA = ChemicalSystem.from_dict(units[0]["inputs"]["stateA"])
+    stateB = ChemicalSystem.from_dict(units[0]["inputs"]["stateB"])
+    ligmap = LigandAtomMapping.from_dict(units[0]["inputs"]["ligandmapping"])
+
+    if any([isinstance(comp, gufe.ProteinComponent) for comp in stateA.components.values()]):
```

Comment:
ToDo: Test if this would still work for files that have been cleaned up, i.e. where components have partially been removed.